### PR TITLE
add UT for linkProgram api

### PR DESCRIPTION
**Pull request by Lukasz-Krakowiak-Mobica**
Add UTs for linkProgram API

### CODE REVIEW

File: tests/test_openclhpp.cpp [hunk 1]
```diff
@@ -3035,5 +3040,98 @@ void testDevicePCIBusInfo_KHR()
...
+    int num_calls)
+{
+    (void) num_calls;
+    TEST_ASSERT_EQUAL_PTR(make_program(0), program);
+    TEST_ASSERT_EQUAL_HEX(CL_PROGRAM_CONTEXT, param_name);
+    TEST_ASSERT(param_value == NULL || param_value_size >= sizeof(cl_context));
+    if (param_value_size_ret != NULL)
+        *param_value_size_ret = sizeof(cl_context);
+    if (param_value != NULL)
+        *(cl_context *) param_value = make_context(0);
```
Comment (keryell) [inline tests/test_openclhpp.cpp:3057]: perhaps this is an opportunity to experiment with more modern style in new code?
```suggestion
    if (param_value != nullptr)
        *static_cast<cl_context *>(param_value) = make_context(0);
```

File: tests/test_openclhpp.cpp [hunk 2]
```diff
@@ -3035,5 +3040,98 @@ void testDevicePCIBusInfo_KHR()
...
+    const cl_device_id * device_list,
+    const char *         options,
+    cl_uint              num_input_programs,
+    const cl_program *   input_programs,
+    void (CL_CALLBACK *  pfn_notify)(cl_program program, void * user_data),
+    void *               user_data,
+    cl_int *             errcode_ret,
+    int                  num_calls)
+{
+    (void) num_calls;
```
Comment (keryell) [inline tests/test_openclhpp.cpp:3072]:
```suggestion
    int                  /* num_calls */)
{
```

Comment (Lukasz-Krakowiak-Mobica) [reply tests/test_openclhpp.cpp:3072]: Done, thanks.

File: tests/test_openclhpp.cpp (hunk 1 above)
Comment (Lukasz-Krakowiak-Mobica) [reply tests/test_openclhpp.cpp:3057]: Done, thanks.

**Review by keryell — APPROVED** [review]
Thanks for the cleaning.

**Review by bashbaug — APPROVED** [review]
LGTM!

**Pull request merged**